.github/workflows/py-checks.yml

```diff
@@ -19,47 +24,64 @@ jobs:
-        python -m isort --profile=black --check --diff --color ./
-    - name: Check with flake8
-      run: |
-        python -m flake8 ./
-    - name: Check with Pylint
-      run: |
-        pylint --recursive=y .
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache pip dependencies for lint/format
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: lint-${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('requirements.dev.txt') }}-${{ github.run_id }}
+          restore-keys: |
+            lint-${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('requirements.dev.txt') }}-
+            lint-${{ runner.os }}-pip-${{ matrix.python-version }}-
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r ./requirements.dev.txt
+      - name: Check with black
+        run: |
+          VERNUM=${{ matrix.python-version }}
+          TARGETV="py${VERNUM/./}"
+          python -m black -t $TARGETV --diff --color --check ./
+      - name: Check imports with isort (black)
+        run: |
+          python -m isort --profile=black --check --diff --color ./
+      - name: Check with flake8
+        run: |
+          python -m flake8 ./
+      - name: Check with Pylint
+        run: |
+          pylint --recursive=y .
+
   pytyping:
     name: Python Type Checks
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       max-parallel: 10
       matrix:
-        python-version: ["3.8", "3.9", "3.10","3.11","3.12"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v3
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r ./requirements.dev.txt
-    - name: Check with MyPy on py${{ matrix.python-version }}
-      run: |
-        mypy --python-version ${{ matrix.python-version }} --warn-unused-ignores ./*.py
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache pip dependencies for typing
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: typing-${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('requirements.dev.txt') }}-${{ github.run_id }}
+          restore-keys: |
+            typing-${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('requirements.dev.txt') }}-
+            typing-${{ runner.os }}-pip-${{ matrix.python-version }}-
```

Comment:
Same here as on lines 38-39.  
Additionally, could we leverage the cache between these jobs?   (So as to avoid caching two copies of pip libs for py 3.10)